### ISSUE TITLE
Partial Search not working

### ISSUE THREAD

**Issue by mandys** [teamtnt/tntsearch#35]
Firstly, thank you for a great plugin.

It's very helpful.

I have everything set up but having one issue with search.

I have 2 emails in the database.

mandy@search.com
mandy2@search.com

If I search for mandy, I only get back mandy@search.com.

Why doesn't mandy2 appear ?

I have fuzziness set to true.
I also have searchBoolean.

Thanks.

**Comment by mandys** [comment]
Okay -

The following works.

2 entries -

mandymoore
mandynomore

search for mandy

both come up.

but mandy@email.com and mandy2@email.com

search for mandy, mandy2 wont come up.

problem seems to be whenever there is a digit in the string.
partial match fails then.

**Comment by nticaric** [comment]
Let me try to explain how it works and why the word `mandy2` won't come up.
Before saving each word to the index, we do a process called stemming, which basically takes the
common root of a word. Since documents are going to use different forms of a word, such as `organize`, `organizes`, and `organizing` the stemming process is needed to be able to give you meaningful results and not to give you an empty result set if you don't know the right word form.

The default stemmer that we use for TNTSearch is the PorterStemmer and if you feed it with the
word `mandy` the stem becomes `mandi`, and if you feed it with `mandy2` it becomes `mandy2`.

So to sum it up, it treats `mandy` and `mandy2` as two different words

**Comment by mandys** [comment]
Thanks for the explanation.

But, that means if someone signs up on the site as jennifer89@email.com, then if i try to search for jennifer i won't get back any results.

**Comment by nticaric** [comment]
That's correct because the text isn't broken down by numbers.  However, you can change this by
implementing your own Tokenizer. Here's the current implementation 

https://github.com/teamtnt/tntsearch/blob/master/src/Support/Tokenizer.php

**Comment by Sciyguy** [comment]
I've got the similar problem, but I am not entirely sure that it has something to do with the current explanation... Taking into account that fuzziness is enabled. 
I've debugged the execution and it seems that some words are simply ignored.
The problem is in the method 'getAllDocumentsForKeyword' (TNTSearch.php:219). In my example 'getWordlistByKeyword' returns 2 words with obviously different ids, but when binding the value, $word[0]['id'] is used, meaning that only one word is used. In my opinion the select statement should search for all of the found words / ids. I didn't look when the tokenizer is applied but at that point, this probably isn't the expected behaviour.

**Comment by nticaric** [comment]
Can you provide a concrete example and I'll try to explain

**Comment by Sciyguy** [comment]
Sure, I have some screenshots, it should be easier. The search phrase is 521 and the keywords saved in TNT db are 521c and 521m. I'm using default fuzzy search config - distance = 2, prefix = 2. 
In my example the second keyword is ignored, because the select statement always returns only the first id. 

![screen shot 2016-09-20 at 06 16 10](https://cloud.githubusercontent.com/assets/896029/18659298/f00c6a78-7f09-11e6-8fb5-4c3f67510e41.png)
![screen shot 2016-09-20 at 06 16 27](https://cloud.githubusercontent.com/assets/896029/18659297/f0088b4c-7f09-11e6-91eb-5cf741664833.png)

**Comment by nticaric** [comment]
@Sciyguy ok, I understand. We should probably add this only when fuzziness is enabled

**Comment by nticaric** [comment]
Closing this one, since the issue should be solved if fuzziness is enabled.
